macros/code-hasher/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Tiny proc macro library designed to hash a code block generating a unique
+//! identifier for it which will get written into a `const` inside of the code
+//! block.
+//!
+//! # Example
+//! extern crate code_hahser;
+//! ```
+//! #[code_hasher::hash(SOME_CONST_NAME, version = "0.1.0")]
+//! pub mod testing_module {
+//!
+//!     pub fn this_does_something() -> [u8; 32] {
+//!         SOME_CONST_NAME
+//!     }
+//! }
+//! ```
+//! Here, `SOME_CONST_NAME` has assigned as value the resulting hash of:
+//! - The code contained inside `testing_module`.
+//! - The version of the crate (code_hasher) version.
+//! - The version passed by the user (is optional). Not adding it will basically
+//!   not hash this attribute and **WILL NOT** use any default alternatives.
+
+use blake3::Hasher;
+use proc_macro::TokenStream;
+
+#[proc_macro_attribute]
+pub fn hash(_attr: TokenStream, _input: TokenStream) -> TokenStream {
+    let mut hasher = Hasher::new();
+    // Add to the hasher the version of the crate used to create the code block
+    // hash.
+    hasher.update(env!("CARGO_PKG_VERSION_MINOR").as_bytes());
```

Comment:
Since you're using the (code) version attribute, you can avoid to pass the crate's version.
I suggested the crate's version only in case of non-generic hasher (so tied to Plonk), when you don't have the (code) version attribute.

Since you implemented, this can be omitted. If, instead, you prefer to add it, please add as mentioned both `major` and `minor`, otherwise if I have `0.4.0` and `1.4.0` it will result in the same hash (and it shouldn't).

Reply:
As discussed this will be removed for now. We can always add it later!